File: test/lit/ts/boolean.ts
// RUN: %cot emit-cir %s | %FileCheck %s

// Phase 1: boolean constants true/false

function returns_true(): boolean {
    return true;
}

function returns_false(): boolean {
    return false;
}

// CHECK-LABEL: func.func @returns_true
// CHECK: cir.constant true : i1
// CHECK: return

// CHECK-LABEL: func.func @returns_false
// CHECK: cir.constant false : i1
// CHECK: return

File: test/lit/swift/boolean.swift
// RUN: %cot emit-cir %s | %FileCheck %s

// libsc: boolean literals and bool function parameters

func yes() -> Bool {
    return true
}

func no() -> Bool {
    return false
}

func identity(flag: Bool) -> Bool {
    return flag
}

// CHECK-LABEL: func.func @yes
// CHECK: cir.constant true : i1
// CHECK: return

// CHECK-LABEL: func.func @no
// CHECK: cir.constant false : i1
// CHECK: return

// CHECK-LABEL: func.func @identity
// CHECK: return %arg0 : i1
